CMakeLists.txt

```diff
@@ -63,8 +63,21 @@ configure_file("${CMAKE_SOURCE_DIR}/cmake/templates/license.h.in" include/licens
 
 include(EthOptions)
 configure_project(TESTS)
-
+set(LATEST_Z3_VERSION "4.8.12")
 find_package(Z3 4.8.0)
+# NOTE: With EMSCRIPTEN=ON we do not build code from test/ dir and that
+# code is our only reason to require latest Z3.
+if (TESTS AND NOT EMSCRIPTEN)
+  if (${Z3_FOUND})
+    if ( NOT ( "${Z3_VERSION_STRING}" VERSION_EQUAL ${LATEST_Z3_VERSION} ) )
```

Comment:
> Xxx_VERSION
The full version string of the package found, if any. Note that many existing modules provide Xxx_VERSION_STRING instead.

I tested Z3_VERSION and Z3_VERSION_STRING with z3 version that provided by ethereum ppa and ubuntu repo. The ubuntu version only provides  Z3_VERSION_STRING.

Reply:
Let's use `Z3_VERSION_STRING` then. It seems to be the variable that [`Z3Config.cmake` generated by Z3](https://github.com/Z3Prover/z3/blob/master/cmake/Z3Config.cmake.in) sets and our custom [`FindZ3.cmake`](https://github.com/ethereum/solidity/blob/develop/cmake/FindZ3.cmake) also sets that variable consistently, unlike `Z3_VERSION`, which it sets only when the stock find script manages to find Z3 on its own.

I actually don't know why we don't consistently set `Z3_VERSION`. Maybe @ekpyron could shed some light here - he added that part in #6891. I think the expectation might have been that it will be set automatically due to CMake's [project()](https://cmake.org/cmake/help/latest/command/project.html) but then I don't understand why we set it when the stock script finds Z3 rather than in the opposite case.